### ISSUE TITLE
torch.nn.softmax sub-optimal performance due to lx scratch optimization issues

### ISSUE THREAD

**Issue by cyang49** [torch-spyre/torch-spyre#2166]
### What happened?

When comparing to torch-sendnn stack, torch-spyre's softmax kernel is about 3x slower in profiling.

There are 2 issues:
1. torch-spyre LX optimization is not turned on by default 
2. Even with optimization turned on, torch-spyre only pinned 1 pair of producer-consumer tensor in a set of ops. Work-division is manually checked and confirmed that the resulting splits are LX optimization friendly (i.e. pinnable in multiple places)

### What did you expect to happen?

- The combined latency of softmax should be close to torch-sendnn stack 
- The compatible splits should result in multiple pinned pairs (should be manually checked)

### How can we reproduce it?

Run profiler calling torch.nn.softmax with tensor shape [1, 512, 4096] and dim=-1

### Any environmental details we need to know?

Use the latest main branches

### Relevant log output

```Shell

```

**Comment by cyang49** [comment]
Waiting for pytorch 2.12 profiling support to remeasure

**Comment by cyang49** [comment]
@chichun-charlie-liu confirmed the torch-spyre softmax (dim=-1) case is properly optimized for the pinnable tensors
